### PR TITLE
#124 Migrated from SharedPreferences to DataStore

### LINKED ISSUE #124: [Refactor]: Implementing DataStore
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the problem

I want to replace sharedPreferences to DataStore

### PR DESCRIPTION
**Pull request by rezazarchi**
Resolves #124 

Migrated from SharedPreferences to DataStore
Converted get functions to return Flow objects for asynchronous data access 
Made set functions suspend functions to ensure proper asynchronous behavior

### CODE REVIEW

File: feature/welcome/src/main/java/com/rahim/yadino/welcome/WelcomeViewModel.kt [hunk 1]
```diff
@@ -19,8 +18,8 @@ class WelcomeViewModel @Inject constructor(
 ) : ViewModel(), WelcomeContract {
 
   private val mutableState = MutableStateFlow<WelcomeContract.WelcomeState>(WelcomeContract.WelcomeState())
-  override val state: StateFlow<WelcomeContract.WelcomeState> = mutableState.onStart {
-    isShowWelcomeScreen()
+  override val state: StateFlow<WelcomeContract.WelcomeState> = isShowWelcomeScreen().map {
+    WelcomeContract.WelcomeState(it)
```
Comment (RahimMahmoudzadeh) [inline feature/welcome/src/main/java/com/rahim/yadino/welcome/WelcomeViewModel.kt:22]: why not use mutableState??!!

please explain to me

File: app/src/cafeBazaar/java/com/rahim/ui/main/MainActivity.kt
Comment (RahimMahmoudzadeh) [inline app/src/cafeBazaar/java/com/rahim/ui/main/MainActivity.kt:1]: please add all change for other flavor (play store & myket)

**Review by RahimMahmoudzadeh — CHANGES_REQUESTED** [review]
Please check the mentioned items

File: feature/welcome/src/main/java/com/rahim/yadino/welcome/WelcomeViewModel.kt (hunk 1 above)
Comment (rezazarchi) [reply feature/welcome/src/main/java/com/rahim/yadino/welcome/WelcomeViewModel.kt]: reverted

File: app/src/cafeBazaar/java/com/rahim/ui/main/MainActivity.kt
Comment (rezazarchi) [reply app/src/cafeBazaar/java/com/rahim/ui/main/MainActivity.kt:1]: done

**Pull request merged**